application/helpers/common_helper.php

```diff
@@ -3999,6 +3999,26 @@ function questionAttributes($returnByName=false)
         "help"=>gT('Use button group or radio list'),
         "caption"=>gT('Display type'));
 
+        /**
+         * New event to allow plugin to add own question attribute (settings)
+         * Using $event->append('questionAttributes', $questionAttributes);
+         * $questionAttributes=[
```

Comment:
I think question attribute should be an object, to enforce that all attributes are defined.